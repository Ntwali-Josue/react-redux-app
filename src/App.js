import React, { Component } from 'react'

class App extends Component {
    constructor (props) {
        super(props);
        this.state = {
            email: "",
            password: "",
        }
    };
    render() {
        return (
            <div>
                <form action="" id= "login">
                    <h1>Sign in</h1>
                    <label htmlFor="">Email</label><br/>
                    <input type="email" placeholder="email"/><br/><br/>
                    <label htmlFor="">Password</label><br/>
                    <input type="password" placeholder="password"/>
                    <br/><br/>
                    <input type="button" value="LOGIN"/>
                </form>
            </div>
        )
    }
}

export default App;
