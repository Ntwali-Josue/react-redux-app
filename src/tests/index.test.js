import React from 'react'
import { render } from "@testing-library/react"
import userEvent from '@testing-library/user-event'
import App from '../App'

const AppComponent = () => {
    return render(<App />)
}
describe("APP", () => {
    it("should render the component", () => {
        const { asFragment } = AppComponent();
        expect(asFragment(<App />)).toMatchSnapshot()
    })
    it("increment", () => {
        const { getByLabelText } = AppComponent();
        const incrementButton = getByLabelText("incrementButton")
        const value = getByLabelText("value")
        const before = parseInt(value.innerHTML, 10);
        userEvent.click(incrementButton);
        const int = parseInt(value.innerHTML);
        expect(int).toBe(before + 1)
    })
    it("decrement", () => {
        const { getByLabelText } = AppComponent();
        const decrementButton = getByLabelText("decrementButton")
        const value = getByLabelText("value")
        const before = parseInt(value.innerHTML, 10);
        userEvent.click(decrementButton)
        const int = parseInt(value.innerHTML);
        expect(int).toBe(before -1)
    })
})