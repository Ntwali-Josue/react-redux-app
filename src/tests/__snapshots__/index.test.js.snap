// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP should render the component 1`] = `
<DocumentFragment>
  <div>
    <h1
      aria-label="value"
      id="value"
    >
      0
    </h1>
    <button
      aria-label="incrementButton"
      id="incrementButton"
      name="increment"
    >
       increment +
    </button>
    <button
      aria-label="decrementButton"
      id="decrementButton"
      name="decrement"
    >
       decrement  -
    </button>
  </div>
</DocumentFragment>
`;
